src/nvim/quickfix.c

```diff
@@ -3319,7 +3319,6 @@ void qf_age(exarg_T *eap)
   }
 
   if (eap->addr_count != 0) {
-    assert(eap->line2 <= INT_MAX);
```

Comment:
this depends on the size of int, which is system-dependent. coverity is complaining because it builds on a system where int=4bytes.

removing this assert will cause the inverse complaint on various static analyzers, e.g. PVS.

Reply:
I am not sure I agree here.
`int` is at least the size of `int32_t` from what I remember.

Furthermore, this check is just done to verify overflow, but if there is actually an overflow here, then the value of `eap->line2` will already be `< INT_MAX`.

My point here is that even if we add the check you mention, this will not change the result: `eap->line2` will always be `< INT_MAX`, even if `SIZEOF_INTMAX_T == SIZEOF_INT32_T`.

Reply:
> int is at least the size of int32_t from what I remember.

`int` can be 2 bytes on old systems. But if our static analyzers don't complain, then yes, let's remove the assert. But I expect the static analyzers will complain. 

We can try it, though.

Reply:
I don't think neovim will ever build on a system that is less than 32bit (but on the other hand, possibly systems where int is bigger than that). I'd rather codify that assumption than jump though a lot of artificial hoops to pretend that we can.

Reply:
Definitely agreed, I just wasn't sure if there's a better way to avoid noise from the analyzers.